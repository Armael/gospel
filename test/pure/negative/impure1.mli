
type t
(*@ mutable model view: int *)

val f: t ->  int
(*@ y = f x
    pure
    modifies x *)

