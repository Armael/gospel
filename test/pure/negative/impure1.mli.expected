
*******************************
********** Parsed file ********
*******************************
type t[@@gospel {| mutable model view: int |}]
val f : t -> int[@@gospel {| y = f x
    pure
    modifies x |}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

type t
  (*@ ephemeral
      model ...
       *)

val f : t -> int
(*@ y = f x
    modifies ...
     *)
File "impure1.mli", line 8, characters 23-24:
Error: Type checking error: a pure function cannot have writes
